src/main/java/com/google/calendar/calendar/handlers/UserHandler.java

```diff
@@ -0,0 +1,47 @@
+package com.google.calendar.calendar.handlers;
+
+import com.google.calendar.calendar.models.User;
+import com.google.calendar.calendar.repositories.UserRepository;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.server.ServerRequest;
+import org.springframework.web.reactive.function.server.ServerResponse;
+import reactor.core.publisher.Mono;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.ConstraintViolationException;
+import javax.validation.Validator;
+
+import java.util.Set;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON;
+
+@Component
+public class UserHandler {
```

Comment:
*A complicação do nosso é código é proporcional a complicação da nossa feature. Quanto mais simples, melhor.*
+
*Avaliamos de forma lógica o nível de complicação de cada trecho de código. Aqui não tem espaço para feeling. Usamos a teoria da carga cognitiva e medimos o nível de pontos de entendimentos necessário por trecho.*

É muito legal que você foi buscar o web flux e desenrolou a funcionalidade inteira com ele. Inclusive usando o driver reativo, o que é animal. Por outro lado, o código final é bem mais complicado do que precisaria. A pessoa tem que sacar de muitas coisas para ficar confortável com esse código. São muitas abstrações novas e não triviais... Para exemplificar: Entender a Mono/Flux e o flatMap(que quase ninguém usa hehe).

Reply:
Nesse caso, o que você recomenda? Deixa de lado o webflux? ou o que seria?

Reply:
Opa, to aqui para facilitar o caminho que você decidir :). Se você, neste momento, se sente capaz de implementar tudo com o webflux, jogue duro! Eu tenho menos vivência com ele, mas considero que domino o suficiente para te colaborar.

Reply:
Beleza vou tentar fazer tudo que tu deixou claro aqui. Valeu pelo retorno rápido.